source/common/network/connection_impl.cc

```diff
@@ -761,6 +762,11 @@ bool ConnectionImpl::bothSidesHalfClosed() {
 }
 
 absl::string_view ConnectionImpl::transportFailureReason() const {
+  // TODO(mattklein123): Is it possible for there to be a local failure reason and a transport
```

Comment:
Maybe make it an ENVOY_BUG if both are set, so we at least know that this is happening.